Tools/scripts/decode-ICSR.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+'''
+decode an stm32 ICSR register value
+'''
+
+import sys
+import optparse
+
+def num(s):
+    try:
+        return int(s)
+    except ValueError:
+        return int(s, 16)
+
+
+parser = optparse.OptionParser(__file__)
+
+opts, args = parser.parse_args()
+
+if len(args) == 0:
+    print(parser.usage)
```

Comment:
usage is missing !

Reply:
It's generated.  I don't really want to fix the usage *right* now - really want to work out how to specify the arch before then.